### PR TITLE
Update token deployer to send chain client to wait for tx mined

### PR DESCRIPTION
**Pull request by kidambisrinivas**
## Motivation

- Update token deployer to send chain client so that ccip integration test clients can wait for token deploy txns to be mined

## Solution

- Use custom function type instead of blockchain.ContractDeployer

### CODE REVIEW

File: integration-tests/ccip-tests/contracts/contract_deployer.go [hunk 1]
```diff
@@ -69,14 +69,28 @@ func (e *CCIPContractsDeployer) DeployLinkTokenContract() (*LinkToken, error) {
 }
 
 func (e *CCIPContractsDeployer) DeployERC20TokenContract(deployerFn blockchain.ContractDeployer) (*ERC20Token, error) {
-	address, _, instance, err := e.evmClient.DeployContract("Custom ERC20 Token", deployerFn)
+	address, tx, _, err := e.evmClient.DeployContract("Custom ERC20 Token", deployerFn)
+	if err != nil {
+		return nil, err
+	}
+
+	err = e.evmClient.ProcessTransaction(tx)
```
Comment (AnieeG) [inline integration-tests/ccip-tests/contracts/contract_deployer.go]: This is already done part of `e.evmClient.DeployContract` you won't need this

File: integration-tests/ccip-tests/contracts/contract_deployer.go [hunk 2]
```diff
@@ -69,14 +69,28 @@ func (e *CCIPContractsDeployer) DeployLinkTokenContract() (*LinkToken, error) {
...
 		return nil, err
 	}
 
 	return &ERC20Token{
 		client:          e.evmClient,
-		instance:        instance.(*erc20.ERC20),
+		instance:        erc20,
 		ContractAddress: *address,
 	}, err
 }
```
Comment (AnieeG) [inline integration-tests/ccip-tests/contracts/contract_deployer.go]: can just `return e.NewERC20TokenContract(*address)`

File: integration-tests/ccip-tests/contracts/contract_deployer.go (hunk 1 above)
Comment (kidambisrinivas) [reply integration-tests/ccip-tests/contracts/contract_deployer.go]: Fixed

File: integration-tests/ccip-tests/contracts/contract_deployer.go (hunk 2 above)
Comment (kidambisrinivas) [reply integration-tests/ccip-tests/contracts/contract_deployer.go]: Fixed

**Review by AnieeG — APPROVED** [review]

**Pull request merged**